### DISCUSSION TITLE
2.4.20 ISO "Fail to enroll could... message authentication failure"

### DISCUSSION THREAD

**Discussion by JMH-Infosec-Forge** [Security-Onion-Solutions/securityonion#11767] [2.4]
### Version

2.4.20

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

4

### RAM

32mb

### Storage for /

64gb

### Storage for /nsm

126gb

### Network Traffic Collection

tap

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

![SO_ErrorMessage_13NOV23](https://github.com/Security-Onion-Solutions/securityonion/assets/139656098/3341d30e-d6ff-45d2-8241-0489e4011e5e)

After the account was built, I received that attached error message:

ERROR: fail to enroll: could not save enrollment information: failed to ensure key: could not get agent key: Cipher: message authentication failed.

I am current using VMware workstation pro 17 on a windows 11 os. The above specs are for the SO server itself (32gb ram, 200gb storage, adapters are VMnet8 = NAT and VMnet1 = host only. 

Can someone point me in the right direction to get this issue resolved. I think I'm close to (FINALLY) spinning up a SO instance on VM after almost a week of troubleshooting and hands on learning. 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by magnoliasock** [comment]
Hi, is the VMnet1 = host only your monitor port?

**Reply by JMH-Infosec-Forge** [reply]
That is correct. I have VMnet8 as my NAT and then VMnet1 as my monitor port... is that supposed to be a different setting?

**Reply by magnoliasock** [reply]
That should be good how you have it.

**Reply by magnoliasock** [reply]
After about a month running my / storage is using 29GB so 64GB should be fine. It's running on bare metal though, my current VM is 2.3 and on vmware 17.5, I'll have to install a new VM and see if I can replicate. I'm curious.

**Reply by JMH-Infosec-Forge** [reply]
I appreciate it... I've definitely had some issues installing. I am using static IPs for both, but I'm thinking if I am running NAT and a host-only and another box as a NAT, I could get away with running DHCP for both VMs.

**Reply by magnoliasock** [reply]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/149980940/98174e83-9e56-49b0-967a-b318645dde7d)
I noticed I had port forwarding on the last one for 443

**Reply by magnoliasock** [reply]
Installing new 2.4 iso now, not going to use port forwarding at this time.

**Reply by magnoliasock** [reply]
Still installing, remembered while doing so some SOC email addresses can cause errors for example clown@clowns.com ; I don't know why but they seem to dislike clowns.com, example.com is safe.

**Reply by JMH-Infosec-Forge** [reply]
I did analyst@root.com

Would it really not recognize that?

**Reply by magnoliasock** [reply]
I'll try adding a user at root.com in my active server, then I can just delete. But I can tell by adding them and if the server is not accessible with that user name-email combo it won't work.

**Reply by magnoliasock** [reply]
@root.com works. That's not it.

**Reply by dougburks** [reply]
> Still installing, remembered while doing so some SOC email addresses can cause errors for example [clown@clowns.com](mailto:clown@clowns.com) ; I don't know why but they seem to dislike clowns.com, example.com is safe.

@magnoliasock I tried installing with clown@clowns.com and it worked fine for me.

**Reply by magnoliasock** [reply]
Hey @dougburks  , I tried again and it worked. About a month ago I added a user with @clowns.com email and he couldn't log in and then changed it to example.com and he could log in no problem. Perhaps it was an issue unrelated to the email address. I thought you just didn't like clowns which is understandable.

**Comment by magnoliasock** [comment]
I had a smooth installation. Did you click 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/149980940/a0748795-837b-4267-b8c4-fbffc6e020df)
before
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/149980940/10c897b2-cded-4bc7-8c1c-96898ee40965)
Trying to think of potential issues.

**Reply by JMH-Infosec-Forge** [reply]
I can try that again and see what happens.

**Comment by dougburks** [comment]
@JMH-Infosec-Forge Did you verify the ISO image as shown at https://securityonion.net/download?

We just released 2.4.30 yesterday, so you might want to download it, verify the download, and then try again.

Have you tried a simple IMPORT installation as shown at https://docs.securityonion.net/en/2.4/first-time-users.html?

How much RAM does your host machine have and is it able to allocate the full 32GB RAM to the VM?

What kind of storage does your host machine have (NVME, SSD, or rotational)?

Is the VM able to connect to the Internet? If not, you might try selecting the airgap option since that will pull everything it needs from the ISO image.

**Reply by JMH-Infosec-Forge** [reply]
I managed to get 2.4.20 to work yesterday.... now I am going back and tracing my steps to see if I can figure out what the disconnect was... 

Have not tried Import as that's not really what I'm looking to do with it.

My host machine has 128gb RAM, I have a SSD for the OS and basic stuff, but the VMs are housed on a 4TB HDD.

What ended up working as doing the "double NAT" approach. I'm not a big fan of this as I wanted to do a closed network, but for the sake of this project I'm doing it might work for now.

**Reply by dougburks** [reply]
> I have a SSD for the OS and basic stuff, but the VMs are housed on a 4TB HDD.

Disk IO could cause problems with Elasticsearch since they typically recommend SSD, so if possible you might want to try a new installation with the VM on the SSD:
https://docs.securityonion.net/en/2.4/hardware.html#elastic-stack
